src/ZODB/mvccadapter.py

```diff
@@ -0,0 +1,268 @@
+"""Adapt IStorage objects to IMVCCStorage
+
+This is a largely internal implementation of ZODB, especially DB and
+Connection.  It takes the MVCC implementation involving invalidations
+and start time and moves it into a storage adapter.  This allows ZODB
+to treat Relstoage and other storages in pretty much the same way and
+also simplifies the implementation of the DB and Connection classes.
+"""
+import threading
+
+import zope.interface
+
+from . import interfaces, serialize, POSException
+from .utils import p64, u64
+
+class Base(object):
+
+    _copy_methods = (
+        'getName', 'getSize', 'history', 'lastTransaction', 'sortKey',
+        'loadBlob', 'openCommittedBlobFile',
+        )
+
+    def __init__(self, storage):
+        self._storage = storage
+        for name in self._copy_methods:
+            if hasattr(self._storage, name):
+                setattr(self, name, getattr(storage, name))
+
+        if interfaces.IBlobStorage.providedBy(storage):
+            zope.interface.alsoProvides(self, interfaces.IBlobStorage)
+
+    def isReadOnly(self):
+        return self._storage.isReadOnly()
```

Comment:
Addressed by changing the copy mechanism: https://github.com/zopefoundation/ZODB/pull/66/commits/c7e99e3de151fee77f6d4c7ac7debe2d2ea4c274